code/modules/mob/living/simple_animal/hostile/roguetown/jitterskull.dm

```diff
@@ -0,0 +1,1349 @@
+    // Critical finisher throttle
+    var/next_critical_available_time = 0 // world.time when we can attempt a critical finisher again
+    // Near-instant attack cadence
+    melee_cooldown = 0.1 SECONDS
+    var/is_feasting = FALSE         // currently lingering and consuming an unconscious target
+    // Anti-stuck tracking
+    var/last_stuck_x = 0
+    var/last_stuck_y = 0
+    var/stuck_ticks = 0
+    // Stalking and aggression flags
+    var/is_stalking = FALSE        // briefly TRUE when we blink away and "stare" before re-engaging
+    var/angry_until = 0            // during this window, skip whiffs and any artificial waiting
+    // Vendetta focus: keep attention on whoever last hurt us for a while
+    var/mob/living/vendetta_target = null
+    var/vendetta_until = 0
+    // Tether throttle to avoid spam teleports
+    var/next_tether_allowed = 0
+    // Guarding: linger around a helpless target before feasting
+    var/is_guarding = FALSE
+    var/mob/living/guarding_target = null
+    var/guarding_until = 0
+    // Searching mode: low alpha while looking for next prey
+    var/is_searching = FALSE
+    var/next_search_announce_time = 0
+    var/no_target_since = 0
+    // After a stalk pause ends, briefly guarantee we actually swing instead of trailing
+    var/primed_attack_until = 0
+    // Track last time we processed post-attack flow to avoid double-handling when mixing signals and fallbacks
+    var/last_attack_flow_at = 0
+
+
+/mob/living/simple_animal/hostile/rogue/jitterskull/electrocute_act(shock_damage, source, siemens_coeff = 1, flags = NONE)
+    return FALSE
+
+/mob/living/simple_animal/hostile/rogue/jitterskull/Initialize()
+    . = ..()
+    set_light(1.5, 1.5, 2, l_color = "#f74242")
+    ADD_TRAIT(src, TRAIT_IGNOREDAMAGESLOWDOWN, TRAIT_GENERIC)
+    ADD_TRAIT(src, TRAIT_NOPAINSTUN, TRAIT_GENERIC)
+    original_alpha = alpha
+    // Run our custom bite logic whenever a melee attack succeeds (works with ai_controller ClickOn flow)
+    RegisterSignal(src, COMSIG_MOB_AFTERATTACK_SUCCESS, PROC_REF(on_after_attack_success))
+    // Make the AI controller move snappier for this pawn
+    if(istype(ai_controller))
+        var/datum/ai_controller/C = ai_controller
+        // Make pathing more responsive so it closes distance quicker
+        C.movement_delay = 0.2 SECONDS
+    // Start idle chatter loop
+    spawn(0)
+        idle_chatter_loop()
+    // Play spawn cinematic and then resume regular behavior
+    spawn(0)
+        spawn_cinematic()
+    // Periodically reacquire humanoid targets if we "forget" them
+    spawn(0)
+        search_for_humanoids_loop()
+    // Anti-stuck: if we're not making progress toward the target, blink closer and continue stalking
+    spawn(0)
+        anti_stuck_stalker_loop()
+    // Persistent pursuit tether: if the target runs far, re-tether near them periodically
+    spawn(0)
+        pursuit_tether_loop()
```

Comment:
Do not do this. `Initialize()` must not sleep and, ideally, shouldn't `spawn()` either. they ESPECIALLY should not `spawn()` infinite loop procs, we have subsystems and AI controllers for a reason.